WFDR-310-AdminEdit/RemoveUsers

setaDM opened the pull request:
## Summary of changes
Changed UI of user, error and admin pages and added delete and edit user button to admin landing page, as well as an autocomplete feature for the itinerary city input box.
## Trello Task link
300 & 310
## Corresponding User Story Number and Title
User story 8: Admin edit user
## Detailed list of changes
Updated UI of:

- adminLanding.html
- addPrefs.html
- addUser.html
- changeAccPass.html
- changePassword.html
- changeUsername.html
- editAccSettings.html
- editPrefs.html
- login.html
- inputEmailForPin.html
- pinConfirmation.html
- error-404.html
- error-405.html
- error-500.html

to match that found in the user landing page.

Added delete and edit user button to admin landing page that work by entering the users ID and clicking the corresponding button. 
Delete user button just removes the user and doesnt change the page.
Edit user button brings you to editUser.html where you can change the information as you'd like.

Created _editUser.html_

- Allows all of a users information to be changed (except for user ID)

Created _autoComplete.js_

- Autocompletes the entered input by city using google maps API

setaDM committed:
Created editUser.html. Added edit user & delete user button to admin page and updated UI

setaDM committed:
Added backend so delete and edit user buttons actually do their intended usage. Added diet tab to table

setaDM committed:
Added feedback for inproper userId, added logo to return back to adminLanding during user edit. Removed some useless imports and minor changes to displayUser in userController

setaDM committed:
Updated UI for user pages and admin page. Created footerBottom.css - static footer at bottom

setaDM committed:
Updated UI for error pages

setaDM committed:
Added footer to error pages, login and signup. Updated UI for forgot password pages

NickMoo5 reviewed (changes requested):
Everything looks like it still works and is functioning. The UI update looks good.

![image](https://github.com/NickMoo5/CMPT-276-Project/assets/78270735/aa6dd7b0-fd7c-45d8-a033-9e18684f690d)
Can we add some padding or margins to separate the table from the footer.

-how much work would it be to sort the list by uid before sending the list of users to the model. 
-Can we remove the admin Account from being displayed? (i.e. remove it from the list before model)

setaDM committed:
Added margin to bottom of error pages and admin user table. Added autocomplete for intinerary city. Added sorting for users by uid. Removed admin from user list on adminLanding page

setaDM commented:
I implemented the changes you requested. Wasn't too much work to add a sorting mechanism for the uid. Lemme know if there is enough spacing from the footer and the table on the admin landing page. I also changed it for the admin edit user page since I noticed it didn't look very good being right underneath the submit button.

NickMoo5 reviewed (approved):
Changes look good and everything working.

setaDM committed:
Added go back button for change username, email, and password page. Changed endpoint for edit preferences to make more sense
